auth/src/main/java/feast/auth/authorization/Keto/KetoAuthorizationProvider.java

```diff
@@ -57,20 +59,28 @@ public KetoAuthorizationProvider(Map<String, String> options) {
    * @return AuthorizationResult result of authorization query
    */
   public AuthorizationResult checkAccess(String project, Authentication authentication) {
-    String email = getEmailFromAuth(authentication);
+    String subject = (String) this.options.get("subject");
+    if ((subject == null) || (subject.isEmpty())) subject = "email";
+    String subjectValue = getSubjectFromAuth(authentication, subject);
     try {
       // Get all roles from Keto
       List<OryAccessControlPolicyRole> roles =
-          this.apiInstance.listOryAccessControlPolicyRoles("glob", 500L, 500L, email);
+          this.apiInstance.listOryAccessControlPolicyRoles("glob", 500L, 500L, subjectValue);
+      List<String> roleTemplates = (List<String>) this.options.get("roles");
 
-      // Loop through all roles the user has
-      for (OryAccessControlPolicyRole role : roles) {
-        // If the user has an admin or project specific role, return.
-        if (("roles:admin").equals(role.getId())
-            || (String.format("roles:feast:%s-member", project)).equals(role.getId())) {
-          return AuthorizationResult.success();
+      for (String roleTemplate : roleTemplates) {
+        if (roleTemplate.contains("{PROJECT}")) {
```

Comment:
`PROJECT` could be a constant and actually could be in a list of constants that could be replaced in the role templates.